ocean_lib/models/test/test_erc721_token.py

```diff
@@ -790,8 +791,47 @@ def test_transfer_nft(web3, config, publisher_wallet, consumer_wallet, factory_r
     assert publish_fees[1] == base_token.address
     assert publish_fees[2] == to_wei(1)
 
+
+def test_nft_transfer_with_pool(
+    web3, config, publisher_wallet, consumer_wallet, factory_router
+):
+    """Tests transferring the NFT before deploying an ERC20, a pool."""
+    erc721_nft, erc20_token = deploy_erc721_erc20(
```

Comment:
'deploy_erc721_erc20()` is called 14 times in this file, typically at the very beginning of a test function. This is a big DRY violation. Rather than asking you to fix it here, I just created https://github.com/oceanprotocol/ocean.py/issues/765.